### ISSUE TITLE
Test Profiler not working

### ISSUE THREAD

**Issue by benthomasknight** [wallabyjs/public#2906]
### Issue description or question

Wallaby Profiling seems to get stuck and does not output anything or open the profiling tool. All I do is click the "Profile" option and select either chrome or VSCode as the output options. The loader at the bottom of VS code runs for a bit then stops with no output. It takes many minutes for it to finish loading.

I have tried manually navigating to the chrome profile page but get a 404.

https://user-images.githubusercontent.com/20388953/149856402-94225698-206e-4924-be8e-3317706a089c.mov

### Wallaby diagnostics report

```
{
  editorVersion: '1.63.2',
  pluginVersion: '1.0.323',
  editorType: 'VSCode',
  osVersion: 'darwin 20.6.0',
  nodeVersion: 'v14.18.2',
  coreVersion: '1.0.1208',
  checksum: 'YmYwNThmM2VjNmJiNDA2ZjRlM2Q3YjllNjliODYxZTMsMTY1Nzg0MzIwMDAwMCww',
  config: {
    files: [
      { pattern: 'lib/**/*.{js,ts}', ignore: false, trigger: true, load: true, instrument: true, order: 1 },
      { pattern: 'test/**/fixtures/*', ignore: false, trigger: true, load: true, instrument: true, order: 2 },
      { pattern: 'test/**/!(*.spec|*.test).ts', ignore: false, trigger: true, load: true, instrument: true, order: 3 },
      { pattern: '*.{ts,js}', instrument: false, ignore: false, trigger: true, load: true, order: 4 }
    ],
    tests: [
      { pattern: 'test/**/*.spec.{js,ts}', ignore: false, trigger: true, load: true, test: true, order: 5 },
      { pattern: 'test/**/*.test.{js,ts}', ignore: false, trigger: true, load: true, test: true, order: 6 }
    ],
    workers: { restart: true, initial: 0, regular: 0, recycle: true },
    env: {
      type: 'node',
      params: {},
      runner: '<homeDir>/.nvm/versions/node/v14.18.2/bin/node',
      viewportSize: { width: 800, height: 600 },
      options: { width: 800, height: 600 },
      bundle: true
    },
    diagnostics: {},
    filesWithNoCoverageCalculated: [],
    runAllTestsInAffectedTestFile: false,
    updateNoMoreThanOneSnapshotPerTestFileRun: false,
    addModifiedTestFileToExclusiveTestRun: true,
    compilers: { '**/*.?(lit)coffee?(.md)': [Function (anonymous)] },
    preprocessors: {},
    maxConsoleMessagesPerTest: 100,
    autoConsoleLog: true,
    delays: { run: 0, edit: 100, update: 0 },
    teardown: undefined,
    hints: {
      ignoreCoverage: '__REGEXP /ignore coverage|istanbul ignore/',
      ignoreCoverageForFile: '__REGEXP /ignore file coverage/',
      commentAutoLog: '?',
      testFileSelection: { include: '__REGEXP /file\\.only/', exclude: '__REGEXP /file\\.skip/' }
    },
    automaticTestFileSelection: true,
    runSelectedTestsOnly: true,
    mapConsoleMessagesStackTrace: false,
    extensions: {},
    reportUnhandledPromises: true,
    testFramework: { version: 'mocha@2.1.0', configurator: 'mocha@2.1.0', reporter: 'mocha@2.1.0', starter: 'mocha@2.1.0' },
    slowTestThreshold: 75,
    lowCoverageThreshold: 80,
    loose: undefined,
    symlinkNodeModules: undefined,
    testLocations: [ 'test/performance.test.ts' ],
    configCode: 'module.exports = function () {\n' +
      '    return {\n' +
      '      files: [\n' +
      "        'lib/**/*.{js,ts}',\n" +
      "        'test/**/fixtures/*',\n" +
      "        'test/**/!(*.spec|*.test).ts',\n" +
      "        { pattern: './*.{ts,js}', instrument: false },\n" +
      '      ],\n' +
      "      tests: ['test/**/*.spec.{js,ts}', 'test/**/*.test.{js,ts}'],\n" +
      '\n' +
      '      workers: {\n' +
      '        restart: true,\n' +
      '      },\n' +
      '\n' +
      '      env: {\n' +
      "        type: 'node',\n" +
      '      },\n' +
      '    };\n' +
      '  };'
  },
  packageJSON: {
    dependencies: {
      '@agyletime/aws-utils': '1.0.1',
      '@agyletime/common-utils': '^0.0.11',
      '@agyletime/mysql-utils': '0.0.16',
      '@sniptt/monads': '0.5.8',
      '@types/aws-lambda': '^8.10.83',
      moment: '^2.29.1',
      'moment-timezone': '^0.5.33',
      mongoose: '^5.13.7',
      'mongoose-createdmodified': '^0.0.2',
      ramda: '^0.26.1'
    },
    devDependencies: {
      '@istanbuljs/nyc-config-typescript': '^1.0.1',
      '@types/chai': '^4.2.22',
      '@types/jsonwebtoken': '^8.5.5',
      '@types/mocha': '^9.0.0',
      '@types/proxyquire': '^1.3.28',
      '@types/ramda': '^0.27.45',
      '@types/sinon': '^10.0.4',
      '@types/sinon-chai': '^3.2.5',
      chai: '^4.3.4',
      codecov: '^3.8.3',
      'dd-trace': '^1.5.0',
      describe: '^1.2.0',
      eslint: '^7.32.0',
      'eslint-config-prettier': '^8.3.0',
      'eslint-plugin-import': '^2.24.0',
      'eslint-plugin-node': '^11.1.0',
      husky: '^7.0.1',
      mocha: '^9.0.3',
      nock: '^13.1.1',
      nyc: '^15.1.0',
      prettier: '^2.3.2',
      'pretty-quick': '^3.1.1',
      proxyquire: '^2.1.3',
      serverless: '^2.13.0',
      'serverless-deployment-bucket': '^1.3.0',
      'serverless-offline': '^8.3.1',
      'serverless-plugin-datadog': '^2.34.0',
      'serverless-plugin-typescript': '^2.1.0',
      'serverless-prune-plugin': '^1.4.3',
      sinon: '^9.2.1',
      'sinon-chai': '^3.7.0',
      'source-map-support': '^0.5.20',
      'ts-node': '^10.3.0',
      typescript: '^4.4.4'
    }
  },
  fs: { numberOfFiles: 39 },
  debug: [
    '2022-01-18T01:41:47.586Z project Wallaby Node version: v14.18.2\n',
    '2022-01-18T01:41:47.587Z project Wallaby config: <homeDir>/at/workactivity/workactivity-metrics-queries/wallaby.conf.js\n',
    '2022-01-18T01:41:48.062Z project File cache: <homeDir>/.vscode/extensions/wallabyjs.wallaby-vscode-1.0.323/projects/c935fb85ee9cd710\n',
    '2022-01-18T01:41:48.167Z uiService Listening port 51235\n',
    '2022-01-18T01:41:48.181Z workers Parallelism for initial run: 10, for regular run: 5\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #0\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #1\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #2\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #3\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #4\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #5\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #6\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #7\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #8\n',
    '2022-01-18T01:41:48.181Z workers Starting run worker instance #9\n',
    '2022-01-18T01:41:48.182Z workers Web server is listening at 56456\n',
    '2022-01-18T01:41:48.287Z project File cache requires some updates, waiting required files from IDE\n',
    '2022-01-18T01:41:48.291Z extended-core New file or complex file change\n',
    '2022-01-18T01:41:48.627Z workers Started run worker instance (delayed) #0\n',
    '2022-01-18T01:41:48.635Z workers Started run worker instance (delayed) #4\n',
    '2022-01-18T01:41:48.635Z workers Started run worker instance (delayed) #3\n',
    '2022-01-18T01:41:48.635Z workers Started run worker instance (delayed) #1\n',
    '2022-01-18T01:41:48.636Z workers Started run worker instance (delayed) #5\n',
    '2022-01-18T01:41:48.636Z workers Started run worker instance (delayed) #2\n',
    '2022-01-18T01:41:48.805Z workers Started run worker instance (delayed) #9\n',
    '2022-01-18T01:41:48.806Z workers Started run worker instance (delayed) #6\n',
    '2022-01-18T01:41:48.808Z workers Started run worker instance (delayed) #8\n',
    '2022-01-18T01:41:48.810Z workers Started run worker instance (delayed) #7\n',
    '2022-01-18T01:41:49.121Z project Stopping process pool\n',
    '2022-01-18T01:41:49.126Z project Running postprocessor\n',
    '2022-01-18T01:41:49.135Z postprocessor New TypeScript language service is required\n',
    '2022-01-18T01:41:49.817Z project Postprocessor execution finished\n',
    '2022-01-18T01:41:49.817Z project Test run started; run priority: 3\n',
    '2022-01-18T01:41:49.818Z project Running all tests\n',
    '2022-01-18T01:41:49.819Z workers Starting test run, priority: 3\n',
    '2022-01-18T01:41:49.819Z nodeRunner Starting sandbox [worker #0, session #pdiuj]\n',
    '2022-01-18T01:41:49.819Z nodeRunner Preparing sandbox [worker #0, session #pdiuj]\n',
    '2022-01-18T01:41:49.819Z nodeRunner Prepared sandbox [worker #0, session #pdiuj]\n',
    '2022-01-18T01:41:49.820Z workers [worker #0, session #pdiuj] Running tests in sandbox\n',
    '2022-01-18T01:41:50.002Z workers [pdiuj] Loaded 1 test(s)\n',
    '2022-01-18T01:41:50.025Z workers [pdiuj] Test executed: succeeds\n',
    '2022-01-18T01:41:50.214Z workers [pdiuj] Run 1 test(s), skipped 0 test(s)\n',
    '2022-01-18T01:41:50.220Z workers [pdiuj] Sandbox is responsive, closing it\n',
    '2022-01-18T01:41:50.220Z workers Starting run worker instance #0\n',
    '2022-01-18T01:41:50.221Z project Test run finished\n',
    '2022-01-18T01:41:50.222Z project Processed console.log entries\n',
    '2022-01-18T01:41:50.222Z project Processed loading sequences\n',
    '2022-01-18T01:41:50.222Z project Processed executed tests\n',
    '2022-01-18T01:41:50.225Z project Processed code coverage\n',
    '2022-01-18T01:41:50.256Z project Test run result processed and sent to IDE\n',
    '2022-01-18T01:41:50.462Z workers Started run worker instance (delayed) #0\n',
    '2022-01-18T01:41:56.267Z uiService UI client connected\n',
    '2022-01-18T01:41:56.267Z uiService Outgoing message ui:handshake\n',
    '2022-01-18T01:41:56.301Z uiService Incoming message ui:tests:resultsRequested\n',
    '2022-01-18T01:41:56.302Z uiService Outgoing message ui:tests:allResultsUpdated\n',
    '2022-01-18T01:41:56.304Z uiService Incoming message ui:start\n',
    '2022-01-18T01:41:56.305Z uiService Outgoing message ui:summary\n',
    '2022-01-18T01:41:56.306Z uiService Outgoing message ui:files\n',
    '2022-01-18T01:42:01.640Z project Requested to run some tests\n',
    '2022-01-18T01:42:01.640Z project Profiler start test run\n',
    '2022-01-18T01:42:01.640Z uiService Outgoing message ui:busy\n',
    '2022-01-18T01:42:01.641Z project Test run started; run priority: 1\n',
    '2022-01-18T01:42:01.641Z testTask Test files from affected: 0, from deleted or manually requested: 1, from recently changed: 0, from loaded by: 0, from failing: 0\n',
    '2022-01-18T01:42:01.641Z testTask Running only selected or not excluded tests\n',
    '2022-01-18T01:42:01.642Z workers Starting test run, priority: 1\n',
    '2022-01-18T01:42:01.642Z nodeRunner Starting sandbox [worker #0, session #v316v]\n',
    '2022-01-18T01:42:01.642Z nodeRunner Preparing sandbox [worker #0, session #v316v]\n',
    '2022-01-18T01:42:01.642Z nodeRunner Prepared sandbox [worker #0, session #v316v]\n',
    '2022-01-18T01:42:01.643Z workers [worker #0, session #v316v] Running tests in sandbox\n',
    '2022-01-18T01:42:01.767Z workers [v316v] Loaded 1 test(s)\n',
    '2022-01-18T01:42:01.778Z workers [v316v] Test executed: succeeds\n',
    '2022-01-18T01:42:02.103Z workers [v316v] Run 1 test(s), skipped 0 test(s)\n',
    '2022-01-18T01:42:02.120Z workers [v316v] Sandbox is responsive, closing it\n',
    '2022-01-18T01:42:02.120Z workers Starting run worker instance #0\n',
    '2022-01-18T01:42:02.348Z workers Started run worker instance (delayed) #0\n',
    '2022-01-18T01:43:03.542Z fs No metadata for added file found: .git/index\n',
    '2022-01-18T01:43:03.542Z fs File changed: .git/index\n',
    '2022-01-18T01:43:03.542Z fs No metadata for changed file found: .git/index\n'
  ]
}
```

**Comment by smcenlly** [comment]
If you try it on a simpler test file (e.g. non asynchronous test, simple assertion, no imports), does it work for you?

We can see that you also have Quokka installed. Can you also please try the following and let us know if it works for you:

1. Run `Quokka.js: New JavaScript File`
2. Set the contents of the file to `const a = 0;`
3. Run `Quokka.js: Profile` and select `Open profile in VS Code`

**Comment by benthomasknight** [comment]
Yeah simple things seem to work just fine. It would be great if this could support fairly complex things as most of what I would like to profile is in slightly more complex situations. Glad to see it is still working though and I am sure I can split things a bit more to make it work.

**Comment by ArtemGovorov** [comment]
@benthomasknight Thanks for checking. It's good to hear it's not completely broken, but rather related to a specific scenario/code path.

Wallaby is using node built-in profiling API to capture the profile, it looks like node may be having some issues recording/serialising it in your case. It may be related to the large profile file size that node can't digest. 

- After you run `Profile` and the Wallaby spinner finally stops (but fails to display any profile view), are there any errors displayed in the `Wallaby Console` output channel?
- Can you please delete `~/.wallaby/test.cpuprofile` file (it there's any file), run profile on your original test again, wait until the Wallaby spinner stops (but fails to display any profile view), and then check if the `~/.wallaby/test.cpuprofile` exists on disk?

> Glad to see it is still working though and I am sure I can split things a bit more to make it work.

If we can get our hands on some repro, we can definitely investigate what's going on.

**Comment by benthomasknight** [comment]
Yeah looks like it is an issue with the tests in the file. I have tried other scenarios and they are all working. I have also tried to pull it out into something I can share but it doesn't seem to want to fail when I do that 🤷 

I checked for the `.cpuprofile` file and couldn't see anything before or after I tried the test.

Oddly I have added the special comment `//?.` and it only takes 10ms to complete so I wouldn't have thought it should be too complex for node. As I can't get this into a state where I can share it I will close this.

Thanks for the help!
